pytest_docker_service/container.py

```diff
@@ -28,14 +28,25 @@ def wait_ready(self):
         Waits for the docker container to be ready and make the test fails if this doesn't happen.
         """
 
+        def _report_last_error(retry_state):
+            pytest.fail(
+                f"failed to get container {self._container.name} ready: {retry_state.outcome.exception()}"
+            )
+
         @tenacity.retry(
             wait=tenacity.wait_fixed(1),
             stop=tenacity.stop_after_delay(10),
-            retry_error_callback=lambda *args: pytest.fail("could not start docker container"),
+            retry_error_callback=_report_last_error,
         )
         def _wait_ready():
             self._container.reload()
-            if not self.ready:
+            if self._container.status == "exited":
+                exit_code = self._container.attrs['State']['ExitCode']
+                raise RuntimeError(
```

Comment:
Wouldn't it be better to return a different error and make sure there's no retry in tenacity on that error? If the container is exited it doesn't make sense to retry anything?